lib/dns.js

```diff
@@ -133,10 +137,24 @@ function onlookupall(err, addresses) {
   }
 }
 
+function refreshDNSCache() {
+  const { DateNow } = primordials;
+  setInterval(() => {
+    for (const hostname in dnsCache) {
+      if (dnsCache[hostname].expiresAt <= DateNow) {
+        delete dnsCache[hostname];
+      }
+    }
+  }, 1000);
+}
+
+refreshDNSCache();
+
 
 // Easy DNS A/AAAA look up
 // lookup(hostname, [options,] callback)
 const validFamilies = [0, 4, 6];
+const dnsCache = {};
```

Comment:
```suggestion
const dnsCache = { __proto__: null };
```
Otherwise a lookup for, say, "constructor" or "toString" results in a cache hit.